predicators/nsrt_learning/strips_learning/base_strips_learner.py

```diff
@@ -46,11 +46,45 @@ def learn(self) -> List[PartialNSRTAndDatastore]:
         if self._verify_harmlessness and not CFG.disable_harmlessness_check:
             logging.info("\nRunning harmlessness check...")
             assert self._check_harmlessness(learned_pnads)
-        min_data = max(CFG.min_data_for_nsrt,
-                       self._num_segments * CFG.min_perc_data_for_nsrt / 100)
-        learned_pnads = [
-            pnad for pnad in learned_pnads if len(pnad.datastore) >= min_data
-        ]
+        # Remove pnads by increasing min_data_perc until harmlessness breaks.
+        if CFG.enable_harmless_op_pruning:
+            assert self._verify_harmlessness
+            assert not CFG.disable_harmlessness_check
+            # Keeps track of latest set of harmless pnads.
+            min_harmless_pnads = learned_pnads
+            # Find the percentage of data in each PNAD uses from lowest
+            # to highest.
+            pnad_perc_data_low_to_high = [
+                len(pnad.datastore) / float(self._num_segments)
+                for pnad in learned_pnads
+            ]
+            pnad_perc_data_low_to_high.sort()
+        else:
+            # If we are not doing harmless operator pruning, return
+            # PNADs at current min_perc_data_for_nsrts.
+            pnad_perc_data_low_to_high = [CFG.min_perc_data_for_nsrt / 100.0]
+        # Iterates over each PNAD in the learned PNADs removing the
+        # PNAD that uses the least amount of data.
+        for min_perc_data_for_nsrt in pnad_perc_data_low_to_high:
+            learned_pnads = self._learn()
+            min_data = max(CFG.min_data_for_nsrt,
```

Comment:
is this going to work as expected if you set `--min_data_for_nsrt` to be something nontrivial, but then also `--enabled_harmless_op_pruning`?

Reply:
It will always chose the max between min_data_for_nsrt and the min_perc_data_for_nsrt. So for all min_perc_data_for_nsrts it is trying below the min_data_for_nsrt it will just use the min_data_for_nsrt instead. If min_data_for_nsrt causes us to break harmlessness it will return the original learned pnads.

Reply:
kk thanks, I think that makes sense